boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -45,29 +49,31 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
             has_charger_faulted = App_Charger_HasFaulted(charger);
         }
 
-        uint8_t    segment = 0U;
-        uint8_t    cell    = 0U;
-        const bool has_reached_max_v =
-            App_Accumulator_GetMaxVoltage(accumulator, &segment, &cell) > MAX_CELL_VOLTAGE_THRESHOLD;
+        const bool has_charging_completed = App_TractiveSystem_GetCurrent(ts) <= CURRENT_AT_MAX_CHARGE;
```

Comment:
We could also OR this with cells have reached max voltage, just in case the charger tries to overcharge the cells. But if that's not a realistic worry, this is fine.